rt/transports/http/src/main/java/org/apache/cxf/transport/http/HTTPConduit.java

```diff
@@ -1427,9 +1432,22 @@ private <T extends Exception> T mapException(String msg,
          * @throws IOException
          */
         protected void handleRetransmits() throws IOException {
+
+            String defaultVerbs = KNOWN_HTTP_VERBS_WITH_NO_CONTENT.stream().collect(Collectors.joining(","));
```

Comment:
@ypiel-talend could you please add `getContextualStrings` to `MessageUtils` class (where we have a set of helper methods to deal with contextual properties):

```
    public static Set<String> getContextualStrings(Message m, String key, Set<String> defaultValue) {
        if (m != null) {
            Object o = m.getContextualProperty(key);
            if (o instanceof String) {
                Set<String> values = new TreeSet<>();
                for (String value : ((String) o).split(",")) {
                    if (!StringUtils.isEmpty(value)) {
                        values.add(value.trim());
                    }
                }
                return values;
            }
        }
        return defaultValue;
    }
```

In this case, your implementation would be just:

```
            final Set<String> allowedVerbsSet = MessageUtils
                    .getContextualStrings(outMessage, AUTHORIZED_REDIRECTED_HTTP_VERBS, 
                        KNOWN_HTTP_VERBS_WITH_NO_CONTENT);
```

Thank you.

Reply:
Thanks, you're right.
Done : https://github.com/apache/cxf/pull/987/commits/af29d1e467a36726d4f8a6f1ee8eab5668f4806e